lib/chef/provider/windows_path.rb

```diff
@@ -0,0 +1,62 @@
+#
+# Author:: Nimisha Sharad (<nimisha.sharad@msystechnologies.com>)
+# Copyright:: Copyright 2008-2017, Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "Win32API" if Chef::Platform.windows?
+require "chef/exceptions"
+
+class Chef
+  class Provider
+    class WindowsPath < Chef::Provider
+      ExpandEnvironmentStrings = Win32API.new("kernel32", "ExpandEnvironmentStrings", %w{ P P L }, "L") if Chef::Platform.windows? && !defined?(ExpandEnvironmentStrings)
+
+      def load_current_resource
+        @current_resource = Chef::Resource::WindowsPath.new(new_resource.name)
+        @current_resource.path(new_resource.path)
+        @current_resource
+      end
+
+      action :add do
+        declare_resource(:env, "path") do
+          action :modify
+          delim ::File::PATH_SEPARATOR
+          value new_resource.path.tr("/", '\\')
+        end
+        ENV["PATH"] = expand_env_vars(ENV["PATH"])
```

Comment:
A comment on why this is necessary might be good, especially since we don't do an equivalent in the `remove` action.